pycode/memilio-epidata/memilio/epidata/modifyDataframeSeries.py

```diff
@@ -70,25 +70,57 @@ def impute_and_reduce_df(
         pass
 
     # range of dates which should be filled
-    if min_date == '':
-        min_date = min(df_old[dd.EngEng['date']])
-    if max_date == '':
-        max_date = max(df_old[dd.EngEng['date']])
+    # to prevent inconsistent data with different start dates, all dates are filled
+    # TODO: find a better method, to prevent unnecessary computation of dates before start date
+    first_date = min(df_old[dd.EngEng['date']])
+    last_date = max(df_old[dd.EngEng['date']])
+
+    # Transform dates to datetime
+    if isinstance(first_date, str) == True:
+        first_date = datetime.strptime(first_date, "%Y-%m-%d")
+    if isinstance(last_date, str) == True:
+        last_date = datetime.strptime(last_date, "%Y-%m-%d")
+
+    # Transform timestamp to date
+    try:
+        first_date = first_date.date()
+        last_date = last_date.date()
+    except:
+        pass
+    # range of dates which should be in output
+    if (min_date is None) or (min_date == ''):
+        min_date = first_date
+    if (max_date is None) or (max_date == ''):
+        max_date = last_date
 
     # Transform dates to datetime
     if isinstance(min_date, str) == True:
         min_date = datetime.strptime(min_date, "%Y-%m-%d")
     if isinstance(max_date, str) == True:
         max_date = datetime.strptime(max_date, "%Y-%m-%d")
 
+    try:
+        min_date = min_date.date()
+        max_date = max_date.date()
+    except:
+        pass
+
     # shift start and end date for relevant dates to compute moving average.
     # if moving average is odd, both dates are shifted equaly.
     # if moving average is even, start date is shifted one day more than end date.
     if moving_average > 0:
-        max_date = max_date + timedelta(int(np.floor((moving_average-1)/2)))
-        min_date = min_date - timedelta(int(np.ceil((moving_average-1)/2)))
-
-    idx = pd.date_range(min_date, max_date)
+        first_date = first_date - timedelta(int(np.ceil((moving_average-1)/2)))
+        last_date = last_date + timedelta(int(np.floor((moving_average-1)/2)))
```

Comment:
Yes. It can go beyond the original dates. For this zero or the first value is backward filled. Without this, the moving average for the first ma/2 days would not be calculated correctly.